### PR TITLE
Bugfix 2x ed25519 verification 2023-11-06

### PR DESCRIPTION
**Pull request by dodmi**

### CODE REVIEW

File: modules/dkimVerifier.jsm [hunk 1]
```diff
@@ -373,6 +373,8 @@ var Verifier = (function() {
 	 */
 	function verifyED25519Sig(key, str, hash_algo, signature, warnings, keyInfo = {}) {
 		let hashedStr = dkim_hash(str, hash_algo, "b64");
+		// Only SHA-256 is allowed, needed to pass tests in verifySignature
+		keyInfo.algName = "sha256";
```
Comment (lieser) [inline modules/dkimVerifier.jsm:377]: Even if maybe now further up only `ed25519-sha256` is allowed, it does not make sense to restrict it here.

**Review by lieser — COMMENTED** [review]
So you got your first ed25519 signed mail? 😁
Note that there are various test mails in https://github.com/lieser/dkim_verifier/tree/master/test/data/dkim. I use this on master for automated test, but maybe they are also helpful for you for some manual testes.

I guess this is to fix the `DKIM_SIGERROR_KEY_HASHMISMATCH` check?
I think it would make more sense to completely delete that check.
On master I already removed it some time ago. Was making some problems and could not remember a good reason why I even added that check.

**Comment by dodmi** [comment]
> So you got your first ed25519 signed mail? 😁 Note that there are various test mails in https://github.com/lieser/dkim_verifier/tree/master/test/data/dkim. I use this on master for automated test, but maybe they are also helpful for you for some manual testes.

I know, that there are a lot of test cases - but it's not easy to test with them in a manual way, since the public keys can't be fetched from DNS. 
I guess, I'll have a look to fake those DNS replies in a test version.

> I guess this is to fix the `DKIM_SIGERROR_KEY_HASHMISMATCH` check? I think it would make more sense to completely delete that check. On master I already removed it some time ago. Was making some problems and could not remember a good reason why I even added that check.

Yes, exactly.
Having a look at it, the RSA verifier of the 2.x version (in contrast to the ED25519 verifier) doesn't take an explicit hash algorithm. So I guess the algorithm is detected in the ASN1 structure of the signature.
So I guess, defining `rsa-sha256` in the DKIM header and adding a `rsa-sha1` signature would circumvent the `SHA1 warning` in the addon. Therefore the exception might be useful.

**Comment by lieser** [comment]
> I know, that there are a lot of test cases - but it's not easy to test with them in a manual way, since the public keys can't be fetched from DNS.
I guess, I'll have a look to fake those DNS replies in a test version.

You can miss use the caching of DKIM keys for this.
Although you can not add manually a key yourself (at least with the GUI in the add-on), you can simply modify an existing one.

**Review by lieser — APPROVED** [review]

**Pull request merged**